File: eBarberShop/eBarberShop.Model/Proizvodi.cs
namespace eBarberShop.Model
{
    public class Proizvodi
    {
        public int ProizvodiId { get; set; }
        public decimal Cijena { get; set; }
        public string Naziv { get; set; }
        public string Sifra { get; set; }
        public string? Opis { get; set; }
        public byte[]? Slika { get; set; }
        public bool? Status { get; set; }
        public int VrstaProizvodaId { get; set; }
        public VrsteProizvoda VrstaProizvoda { get; set; }

        public string? VrstaProizvodaNaziv => VrstaProizvoda?.Naziv;
    }
}

File: eBarberShop/UI/ebarbershop_desktop/lib/models/proizvodi/proizvodi.dart
import 'package:json_annotation/json_annotation.dart';

part 'proizvodi.g.dart';

@JsonSerializable()
class Proizvodi {
  int proizvodiId;
  double cijena;
  String naziv;
  String sifra;
  String? opis;
  String? slika;
  bool? status;
  int vrstaProizvodaId;
  String? vrstaProizvodaNaziv;

  Proizvodi(this.proizvodiId, this.cijena, this.naziv, this.sifra, this.opis, this.slika, this.status, this.vrstaProizvodaId, this.vrstaProizvodaNaziv);

  factory Proizvodi.fromJson(Map<String, dynamic> json) =>
      _$ProizvodiFromJson(json);

  Map<String, dynamic> toJson() => _$ProizvodiToJson(this);
}